### ISSUE TITLE
`sudo make install` leaves a `doc/help` directory owned by root in the source tree

### ISSUE THREAD

**Issue by sanjayankur31** [nest/nest-simulator#574]
Description of problem:
-----------------------------

Running `make install` to install nest to the expected directory, such as `/opt` leaves a `doc/help` directory owned by the `root` user in the source tree. 

Version-Release or Git commit:
----------------------------------------

```
[asinha@ankur  nest-simulator(master %=)]$ git lg | head -1
*   782d30a - (2 days ago) Merge pull request #573 from lekshmideepu/travis_build_log_fix — Jochen Martin Eppler 
```

How reproducible: (Always/Sometimes/Unsure)
------------------------------------------------------------

Always

Steps to Reproduce:
---------------------------

1. Run cmake:

```
INSTALL_PATH="/opt/"
cmake -DCMAKE_INSTALL_PREFIX:PATH=$INSTALL_PATH -Dwith-python:BOOL=ON -Dwith-mpi:BOOL=ON .
```

2. Run make and make install

```
make -j6; sudo make install; 
```

Actual results:
------------------

A `doc/help` directory that is owned by `root` is left behind after the `sudo make install`.

Expected results:
----------------------

Directory should not be left behind, and certainly shouldn't be owned by root? Only noticed it when `git clean -dfx` failed to completely clear the source tree.

Additional info:
-------------------
Directory is not created after `make`, it only turns up after `make install`:

```
[asinha@ankur  nest-simulator(master %=)]$ pwd
/home/asinha/Documents/02_Code/00_repos/01_others/nest-simulator

[asinha@ankur  nest-simulator(master %=)]$ lash /opt/share/doc/nest/help/ ./doc/help/
./doc/help/:
total 348K
 36K drwxr-xr-x. 2 root root  36K Dec  3 11:22 cc
 72K -rw-r--r--. 1 root root  71K Dec  3 11:22 helpindex.hlp
172K -rw-r--r--. 1 root root 172K Dec  3 11:22 helpindex.html
 68K drwxr-xr-x. 2 root root  64K Dec  3 11:22 sli

/opt/share/doc/nest/help/:
total 348K
 36K drwxr-xr-x. 2 root root  36K Dec  3 11:22 cc
 72K -rw-r--r--. 1 root root  71K Dec  3 11:22 helpindex.hlp
172K -rw-r--r--. 1 root root 172K Dec  3 11:22 helpindex.html
 68K drwxr-xr-x. 2 root root  68K Dec  3 11:22 sli
```

**Comment by heplesser** [comment]
@steffengraber Could you take a look at this? I see two solutions: 
- generate help as part of `make`
- still generate help in `make install`, but run the process building the pages with normal user privileges; I guess that would be safer, too.

**Comment by steffengraber** [comment]
The documentation says:  

    cmake -DCMAKE_INSTALL_PREFIX:PATH=</install/path> </path/to/NEST/src>
    make
    make install
 
For me, with this, 'git clean -dfx' works fine.

Installing as root do not only creates the help files with su priviliges. There are some more in different folders. 
So, if you make 'sudo make install', you have to 'sudo git clean -dfx' to clear the source tree.

@heplesser  Maybe, it is enough to adapt the INSTALL doc to explain this a little bit?

**Comment by sanjayankur31** [comment]
Usually, (and I'd think that this is more a convention than a rule) the `make install` step only copies files to the `INSTALL_PATH`, the building process is done in the `make` step. The issue is that system wide installation either in the FHS folders or alternatives such as `/opt/` or `/usr/local` etc. requires the use of `sudo make install` - non root users cannot write to these directories. For the time being, running `sudo git clean ...` works, but it must be used with caution, of course :)

**Comment by jougs** [comment]
@sanjayankur31: would you be willing to review #647 and comment here if that solves the problem for you? Thanks!

**Comment by heplesser** [comment]
Fixed by merge of #647.
